starknet_py/tests/e2e/client/client_test.py

```diff
@@ -293,6 +311,15 @@ async def test_deploy(balance_contract, gateway_client):
     assert result.code == "TRANSACTION_RECEIVED"
 
 
+@pytest.mark.asyncio
+async def test_declare(balance_contract, gateway_client):
+    # TODO extend this test to all clients
+    declare_tx = make_declare_tx(compiled_contract=balance_contract)
+    result = await gateway_client.declare(declare_tx)
+
+    assert result.code == "TRANSACTION_RECEIVED"
+
+
```

Comment:
This will duplicate test we have in #266, so before we merge this should be removed.